src/control/Control.cpp

```diff
@@ -1235,6 +1238,48 @@ void Control::deletePage() {
 
 void Control::insertNewPage(size_t position) { pageBackgroundChangeController->insertNewPage(position); }
 
+void Control::appendNewPdfPages() {
+    int pageCount = this->doc->getPageCount();
+    // find last page with pdf background and get its pdf page number
+    PageRef page;
+    int lastPdfPage = -1;
+    for (int i = pageCount - 1; i >= 0; --i) {
+        page = doc->getPage(i);
+        if (page && page->getBackgroundType().isPdfPage()) {
+            lastPdfPage = page->getPdfPageNr();
+            break;
+        }
+    }
+    if (lastPdfPage == -1) {
+        string msg = FS(_F("Document has no pdf pages."));
+        XojMsgBox::showErrorToUser(getGtkWindow(), msg);
+        return;
+    }
```

Comment:
I think in this cornercase should we should add the new pages. Imagine a document with all PDF pages deleted, and you add new pages to the PDF, now all the pdfs should be added. Am I right?

```suggestion
```

Reply:
Ok, I am fine with adding all pdf pages in that case. If all pdf pages are deleted and you save/reload the document there won't be any pdf pages to add, though. It only occurs if you remove the last pdf page in the document and don't save.